### PR TITLE
Simplify `SlackEventApiEndpointView` methods

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does

Simplifies `_get_organization_from_payload` and `_get_slack_team_identity_from_payload` methods on `SlackEventApiEndpointView`, so it's (hopefully) easier to grasp.

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/slack/views.py [hunk 1]
```diff
@@ -79,8 +82,6 @@
 
 logger = logging.getLogger(__name__)
 
-SELECT_ORGANIZATION_AND_ROUTE_BLOCK_ID = "SELECT_ORGANIZATION_AND_ROUTE"
```
Comment (vadimkerr) [inline engine/apps/slack/views.py:82]: I couldn't find any usages, so removing it.

File: engine/apps/slack/views.py [hunk 2]
```diff
@@ -433,115 +433,93 @@ def post(self, request):
...
-            return slack_team_identity
+            with suppress(KeyError):
+                return payload["team_id"]
+
+            return None
 
         try:
-            slack_team_identity = SlackTeamIdentity.objects.get(slack_id=slack_team_id)
-        except SlackTeamIdentity.DoesNotExist as e:
-            logger.warning("Team identity not detected, that's dangerous!" + str(e))
```
Comment (vadimkerr) [inline engine/apps/slack/views.py:449]: Already logged [here](https://github.com/grafana/oncall/pull/2384/files#diff-1784f1d0d65fa477f4562e73aa23fe1c757b171f36e03f12600bdb021f121307R194-R196)

**Review by joeyorlando — APPROVED** [review]
LGTM 👍 

Up to you, but it might be useful adding a few small unit tests for `_get_organization_from_payload` and `_get_slack_team_identity_from_payload` to ensure they handle all the possible payload scenarios as you'd expect.

**Comment by vadimkerr** [comment]
@joeyorlando Thank you for the review! I'm currently working on https://github.com/grafana/oncall/issues/2336 and it will introduce a few changes to the Slack view, so I'm planning to add tests on most of this logic in subsequent PRs.

**Pull request merged**